packages/node/src/store.ts

```diff
@@ -263,4 +264,33 @@ export class Store {
     );
     return await this.getStateChannel(multisigAddress);
   }
+
+  async setCommitmentForAppInstanceHash(
+    appInstanceHash: string,
+    protocol: string,
+    commitment: string
+  ) {
+    const key = this.computeCommitmentKey(appInstanceHash, protocol);
+    const value = JSON.stringify(commitment);
+    await this.storeService.set([
+      {
+        key,
+        value
+      }
+    ]);
+  }
+
+  async getCommitmentForAppInstanceHash(
+    appInstanceHash: string,
+    protocol: string
+  ): Promise<string> {
+    const key = this.computeCommitmentKey(appInstanceHash, protocol);
+    return this.storeService.get(key);
+  }
+
+  private computeCommitmentKey(appInstanceHash: string, protocol: string) {
```

Comment:
I think we can type check `protocol` here to be in an enum, probably one already defined in the `types`